app/src/main/java/com/alexbar/layoutbasic/movies_app/utils/DataStoreManager.kt

```diff
@@ -0,0 +1,62 @@
+package com.alexbar.layoutbasic.movies_app.utils
+
+import android.content.Context
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.booleanPreferencesKey
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.core.emptyPreferences
+import androidx.datastore.preferences.preferencesDataStore
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.map
+import java.io.IOException
+
+class DataStoreManager(private val context: Context) {
+
+    companion object {
+        private val Context.dataStore: DataStore<Preferences> by preferencesDataStore(name = MovieConstants.data_store_key)
+        val showMoviesKey = booleanPreferencesKey(MovieConstants.show_movies_key)
+        val showSeriesKey = booleanPreferencesKey(MovieConstants.show_series_key)
+    }
+
```

Comment:
In my code, I always use one class for setting values and one class for getting values. Then, I can connect all my setters and getters to a main function in these classes ,like :

 ```
   private suspend fun setPreference(key: Preferences.Key<Boolean>, value: Boolean) {
        context.dataStore.edit { preferences ->
            preferences[key] = value
        }
    }
```

```
    private fun getPreference(key: Preferences.Key<Boolean>, defaultValue: Boolean): Flow<Boolean> {
        return context.dataStore.data
            .catch { exception ->
                if (exception is IOException) {
                    emit(emptyPreferences())
                } else {
                    throw exception
                }
            }
            .map { preferences ->
                preferences[key] ?: defaultValue
            }
    }
```